### DISCUSSION TITLE
maturin build/develop does not generate stub-files

### DISCUSSION THREAD

**Discussion by muellevin** [PyO3/maturin#1818] [Q&A] [answered]
Hello, I am currently testing a project to incorporate Rust functionality into Python. At the moment, I am facing issues with auto-completion for PyO3 and CFFI interface generation. Is there a way to generate stub files in the Maturin development process for these libraries without the need for manual additions?

I am working on an conda environment on Windows and i am using Visual Studio Code. 

I appreciate any assistance you can provide. Thank you.

**Comment by messense** [comment, marked as answer]
> Is there a way to generate stub files in the Maturin development process for these libraries without the need for manual additions?

No. For pyo3 see https://github.com/PyO3/pyo3/issues/2454, as for CFFI I don't know.
